opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -94,8 +95,11 @@ public void finish() {
 
     @Override
     public synchronized void finish(long finishMicros) {
-        this.finishMicros = finishMicros;
-        this.mockTracer.appendFinishedSpan(this);
+        if (!finished) {
```

Comment:
I'm doing this change because span was added multiple times to `Tracer#finishedSpans`

> I'd argue that this should at least log something in the finished==true case.

Agree, this can help find issues in instrumentation. Log statement has been added. 

Actually why is not `Span.finish()` idempotent by definition? Is there any reason for it? If multiple calls on `finish()` are not defined then behavior from tracer to tracer might by different, then it could be more difficult to change a tracing system.